### ISSUE TITLE
A little confusion from a newbie.How to get the loss function of REINFORCE?

### ISSUE THREAD

**Issue by natpagle204** [JamesChuanggg/pytorch-continuous-REINFORCE#1]
Hi, thanks for your code, they are well written and very helpful to newbie like me!!

But there is one line that I don't understand, I'd appreciate if you could explain that to me.

loss = loss - (log_probs[i]*(Variable(R).expand_as(log_probs[i])).cuda()).sum() - (0.0001*entropies[i].cuda()).sum()
loss = loss / len(rewards)

![image](https://user-images.githubusercontent.com/30690978/29492800-0c03ac20-85ba-11e7-879c-ddae7c3dbc8a.png)

I wonder how can I derive the loss function of REINFOCE algorithm. And why does the entropy term appear? This doesn't seem to be straight forward to me.

**Comment by chingyaoc** [comment]
Hi, 
This is the update rule for policy gradient method.
I recommend you to check Sutton (a.k.a godfather of RL)'s [paper](https://papers.nips.cc/paper/1713-policy-gradient-methods-for-reinforcement-learning-with-function-approximation.pdf).
Entropy term is a common trick that encourages policy to be less deterministic which helps exploration.
Hope these informations help you :)
